frontend/lib/Views/AdminPage/groups/add_user_modal.dart

```diff
@@ -0,0 +1,133 @@
+import 'package:adpv_frontend/DataModels/member_summary.dart';
+import 'package:adpv_frontend/Views/AdminPage/utils.dart';
+import 'package:flutter/material.dart';
+
+import '../../../Repository/AdminRepository/admin_gateway.dart';
+import '../../../Widgets/common_widgets.dart';
+
+MaterialStateProperty<EdgeInsetsGeometry?> buttonPadding =
+    MaterialStateProperty.all(const EdgeInsets.all(20));
+
+void showAddUserModal(
+  BuildContext context,
+  AdminGateway gateway,
+  int groupId,
+  Function(String email) onProceedFunction,
+) {
+  showDialog(
+    context: context,
+    builder: (_) {
+      final FocusNode focusNode = FocusNode();
+      final GlobalKey autocompleteKey = GlobalKey();
+      final TextEditingController emailController = TextEditingController();
+      return AlertDialog(
+        title: const Text('Add member'),
+        content: FutureBuilder<List<UserSummary>>(
+          future: gateway.getMembersNotInGroup(groupId),
+          builder: (context, snapshot) {
+            if (snapshot.connectionState == ConnectionState.none ||
+                snapshot.data == null) {
+              return loadingInCenter();
+            } else {
+              return SizedBox(
+                width: MediaQuery.of(context).size.width * 0.7,
+                height: MediaQuery.of(context).size.width * 0.2,
+                child: Scaffold(
+                  appBar: AppBar(
+                    shadowColor: Colors.transparent,
+                    elevation: 0,
+                    backgroundColor: Colors.white,
+                    automaticallyImplyLeading: false,
+                    title: TextFormField(
+                      controller: emailController,
+                      focusNode: focusNode,
+                      decoration: const InputDecoration(
+                        hintText: "User's email",
+                      ),
+                      onFieldSubmitted: (String value) {
+                        RawAutocomplete.onFieldSubmitted<String>(
+                          autocompleteKey,
+                        );
+                      },
+                    ),
+                  ),
+                  body: RawAutocomplete<String>(
+                    key: autocompleteKey,
+                    textEditingController: emailController,
+                    focusNode: focusNode,
+                    optionsBuilder: (TextEditingValue textEditingValue) {
+                      if (textEditingValue.text == '') {
+                        return castToEmails(snapshot.data!);
+                      }
+                      return castToEmails(snapshot.data!).where(
+                        (user) => user
+                            .toLowerCase()
+                            .contains(textEditingValue.text.toLowerCase()),
+                      );
```

Comment:
Czy ten `if` jest tutaj potrzebny? Skoro jeszcze nic nie wpisaliśmy, to chyba powinien nam pokazać wszystkie adresy email, bo sprawdzamy, czy w liście maili mamy string'a, który jest pusty